walkthroughs/howto-mutual-tls-acm/README.md

```diff
@@ -0,0 +1,277 @@
+# Configuring Mutual TLS with ACM PCA TLS Certificates
+
+In this walkthrough, like the [basic file-based TLS example](../howto-tls-file-provided), we'll enable TLS encryption with mutual (two-way) authentication between two endpoints in App Mesh using X.509 certificates.
+
+## Introduction
+
+In App Mesh, traffic encryption is originated and terminated by the Envoy proxy. This means your application code is not responsible for negotiating a TLS-encrypted connection, instead allowing the local proxy to negotiate and terminate TLS on your application's behalf.
+
+In a basic TLS encryption scenario (for example, when your browser originates an HTTPS connection), the server would present a certificate to any client. In Mutual TLS, both the client and the server present a certificate to each other, and both validate the peer's certificate.
+
+Validation typically involves checking at least that the certificate is signed by a trusted Certificate Authority, and that the certificate is still within its validity period. The application (in our case, Envoy) may also choose to validate other aspects of the certificate, such as the Subject Alternative Name identity signed into the certificate.
+
+In this guide, we will be configuring Envoy proxies using certificates hosted in AWS Secrets Manager, which a modified Envoy image will retrieve during startup. We will have a virtual gateway connected to a single backend service. Both the gateway and backend proxies will present certificates signed by the same Certificate Authority (CA), though you could choose to use separate CAs.
+
+## Prerequisites
+
+1. Install Docker. It is needed to build the demo application images.
+2. Install the unix command line utility `jq`. If you don't already have it, [you can install it from here](https://stedolan.github.io/jq/).
+3. Install session manager plugin from [here](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html). If that is not possible, use the AWS console to run commands from bastion host.
+
+## Part 1: Setup
+
+### Step 1: Create Color App Repository
+
+We'll start by setting up the basic docker images for our services. All commands will be provided as if run from the same directory as this README.
+
+First, we need to set a few environment variables before provisioning the infrastructure. Please change the value for `AWS_ACCOUNT_ID` and `ENVOY_IMAGE` below.
+
+```bash
+export AWS_ACCOUNT_ID=<your account id>
+export AWS_DEFAULT_REGION=us-west-2
+export ENVIRONMENT_NAME=mtls-ec2-appmesh-example
+export MESH_NAME=mtls-ec2-appmesh
```

Comment:
is this an EC2 only solution or does is also work in FARGATE?

Reply:
This solution is for ECS Fargate but can run on any compute that supports containers like AWS ECS on EC2, AWS EKS on EC2 or AWS EKS on Fargate